### PR TITLE
k8s version store in cluster, available in app detail and cluster get api's

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

on cluster create and update operation, fetch and persist kubernetes server version. it will be used in app detail page for showing correct version specific grafana graphs.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Cluster Create
- [x] Cluster Update
- [x] Cluster Get
- [x] App Detail Get
- [x] App Detail - Graphs specific to  K8s Version.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have tested it for all user roles

### CODE REVIEW

File: internal/util/K8sUtil.go [hunk 1]
```diff
@@ -47,6 +48,20 @@ func (impl K8sUtil) GetClient(clusterConfig *ClusterConfig) (*v12.CoreV1Client,
 	cfg.BearerToken = clusterConfig.BearerToken
 	cfg.Insecure = true
 	client, err := v12.NewForConfig(cfg)
+
+	return client, err
+}
+
+func (impl K8sUtil) GetK8sVersion(clusterConfig *ClusterConfig) (*discovery.DiscoveryClient, error) {
```
Comment (nishant-d) [inline internal/util/K8sUtil.go:55]: method name is misleading

File: pkg/cluster/ClusterService.go [hunk 2]
```diff
@@ -81,21 +84,47 @@ type ClusterServiceImpl struct {
...
 		logger:                         logger,
 		environmentRepository:          environmentRepository,
 		grafanaClient:                  grafanaClient,
 		installedAppRepository:         installedAppRepository,
 		clusterInstalledAppsRepository: clusterInstalledAppsRepository,
+		K8sUtil:                        K8sUtil,
 	}
 }
 
+const ClusterName = "default_cluster"
```
Comment (nishant-d) [inline pkg/cluster/ClusterService.go:104]: probably duplicate code

**Review by nishant-d — APPROVED** [review]

File: internal/util/K8sUtil.go (hunk 1 above)
Comment (vikramdevtron) [reply internal/util/K8sUtil.go]: changed to GetK8sDiscoveryClient

File: pkg/cluster/ClusterService.go (hunk 2 above)
Comment (vikramdevtron) [reply pkg/cluster/ClusterService.go:105]: removed from environment service.

File: pkg/app/AppListingService.go [hunk 3]
```diff
@@ -536,6 +543,13 @@ func (impl AppListingServiceImpl) FetchAppDetails(appId int, envId int) (bean.Ap
 
 	appDetailContainer.LinkOuts = linkouts
 	appDetailContainer.AppId = appId
+
+	envModel, err := impl.environmentRepository.FindById(envId)
+	if err != nil {
```
Comment (pghildiyal) [inline pkg/app/AppListingService.go:548]: shouldnt we handle norowerr separately?

File: pkg/cluster/ClusterService.go [hunk 4]
```diff
@@ -352,6 +402,22 @@ func (impl ClusterServiceImpl) Update(bean *ClusterBean, userId int32) (*Cluster
 	model.Config = bean.Config
 	model.UpdatedBy = userId
 	model.UpdatedOn = time.Now()
+
+	if len(model.K8sVersion) == 0 {
```
Comment (pghildiyal) [inline pkg/cluster/ClusterService.go:406]: ppl can update cluster version, what ll happen in that case?

File: pkg/cluster/ClusterService.go [hunk 5]
```diff
@@ -127,6 +157,20 @@ func (impl ClusterServiceImpl) Save(bean *ClusterBean, userId int32) (*ClusterBe
 	model.UpdatedBy = userId
 	model.CreatedOn = time.Now()
 	model.UpdatedOn = time.Now()
+
+	cfg, err := impl.GetClusterConfig(bean)
+	if err != nil {
```
Comment (pghildiyal) [inline pkg/cluster/ClusterService.go:162]: shouldnt we handle norowerr separately?

**Review by nishant-d — APPROVED** [review]

**Pull request merged**